src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const logs = [];
+  let currentState = state;
+
+  for (const action of actions) {
+    let stateCopy = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        if ('extraData' in action) {
+          Object.assign(stateCopy, action.extraData);
+        }
```

Comment:
Not sure you need this if statement, we always have `extraData` for this type of action